modules/access_affinitygroup/access_affinitygroup.module

```diff
@@ -852,6 +852,35 @@ function access_affinitygroup_user_login(UserInterface $account) {
   }
 }
 
+/**
+ * Implements hook_user_presave().
+ * If user's email address changes, tell simplelists email.
+ */
+function access_affinitygroup_user_presave($uu) {
+
+  if ($uu->original->getEmail() <> $uu->getEmail()) {
+    $msg = "";
+    $simpleListsApi = new SimpleListsApi();
+    \Drupal::logger('access_affinitygroup')->debug("User email changed: " . $uu->getEmail() . ' from ' . $uu->original->getEmail());
+    $status = $simpleListsApi->updateUserEmailAddress($uu->original->getEmail(), $uu->getEmail(), $msg);
+
+    // email not found in simplelists account.
+    if (!isset($status)) {
+      return;
+    }
+
+    // User is a contact in simplelists. Change email address.
+    $displayMsg =  $uu->getEmail() . ' from ' . $uu->original->getEmail() . ' (' . $uu->id() . ')';
+    if ($status) {
+      $displayMsg = "SimpleLists email changed.  $displayMsg";
```

Comment:
Did you mean to concatenate the $displayMsg?

Reply:
Yes!

Reply:
Because it is for logging. We can have a record to look at  if things are weird on the simplelists account.